packages/zosfiles/src/utils/ZosFilesHeaders.ts

```diff
@@ -0,0 +1,268 @@
+/*
+* This program and the accompanying materials are made available under the terms of the
+* Eclipse Public License v2.0 which accompanies this distribution, and is available at
+* https://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Copyright Contributors to the Zowe Project.
+*
+*/
+
+import { IHeaderContent } from "@zowe/imperative";
+import { ZosmfHeaders } from "@zowe/core-for-zowe-sdk";
+
+/**
+ * Enumeration of operation contexts used when generating content-type headers.
+ *
+ * These different contexts along with passed in options are used to apply the appropriate
+ * content-type header, with the default being: { "X-IBM-Data-Type": "text" }
+ */
+export enum ZosFilesContext {
+    USS_MULTIPLE = "uss_multiple", //general content-header (json)
+    DOWNLOAD = "download", // options.localEncoding matters
+    UPLOAD = "upload", // might add an octet-stream header, options.encoding matters
+    ZFS = "zfs", //no content-headers
+    LIST = "list", //no content-headers
+}
+
+/**
+ * This class centralizes REST request headers creation logic across all ZosFiles methods.
+ */
+export class ZosFilesHeaders {
+    // ============================//
+    // INITIALIZATION & HEADER MAP //
+    // ============================//
+
+    /**
+     * Initializes the header map with predefined header generation functions.
+     * To extend header generation, add new keys and functions here.
+     */
+    private static headerMap = new Map<string, <T>(options: T, context?: ZosFilesContext) => IHeaderContent | IHeaderContent[]>();
+    static initializeHeaderMap() {
+        this.headerMap.set("binary", (options) => (options as any).binary === true ? ZosmfHeaders.X_IBM_BINARY : undefined);
+        this.headerMap.set("record", (options) => (options as any).binary !== true ? ZosmfHeaders.X_IBM_RECORD : undefined);
```

Comment:
The system tests are failing because many data sets are being uploaded with a record header. This line seems to be setting the record header if the data set is not being transferred in binary mode. I believe that instead it should only be setting the record header if the record option is true.